src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1359,75 +1216,24 @@ private static bool IsValidCallback(MethodInfo method, ParameterInfo[] parameter
                     return _knownDataContracts;
                 }
 
-                set
-                { _knownDataContracts = value; }
-            }
-
-            internal override bool IsISerializable
-            {
-                get { return _isISerializable; }
-                set { _isISerializable = value; }
-            }
-
-            internal bool HasDataContract
-            {
-                get { return _hasDataContract; }
-            }
-
-            internal bool HasExtensionData
-            {
-                get { return _hasExtensionData; }
-                set { _hasExtensionData = value; }
-            }
-
-            internal bool IsNonAttributedType
-            {
-                get { return _isNonAttributedType; }
+                set { _knownDataContracts = value; }
             }
 
-            [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
-            private void SetKeyValuePairAdapterFlags(
-                [DynamicallyAccessedMembers(DataContractPreserveMemberTypes)]
-                Type type)
-            {
-                if (type.IsGenericType && type.GetGenericTypeDefinition() == Globals.TypeOfKeyValuePairAdapter)
-                {
-                    _isKeyValuePairAdapter = true;
-                    _keyValuePairGenericArguments = type.GetGenericArguments();
-                    _keyValuePairCtorInfo = (ConstructorInfo)type.GetMemberWithSameMetadataDefinitionAs(s_ctorGenericMethod);
-                    _getKeyValuePairMethodInfo = (MethodInfo)type.GetMemberWithSameMetadataDefinitionAs(s_getKeyValuePairMethod);
-                }
-            }
+            internal string? SerializationExceptionMessage => _serializationExceptionMessage;
 
-            private bool _isKeyValuePairAdapter;
-            private Type[]? _keyValuePairGenericArguments;
-            private ConstructorInfo? _keyValuePairCtorInfo;
-            private MethodInfo? _getKeyValuePairMethodInfo;
+            internal string? DeserializationExceptionMessage => (_serializationExceptionMessage == null) ? null : SR.Format(SR.ReadOnlyClassDeserialization, _serializationExceptionMessage);
 
-            internal bool IsKeyValuePairAdapter
+            internal override bool IsISerializable
             {
```

Comment:
Make this an auto property?

Reply:
The backing field is used in the constructor. I don't think there would be any issue using the property in the constructor since it's just a bool, but... do we have guidance on preferred form here?

Reply:
I don't know about in the runtime repo, but I know VS will bug me about this as a potential code fix so all through WCF and CoreWCF I set auto properties in constructors.